mzLib/Omics/Fragmentation/Product.cs

```diff
@@ -89,14 +89,18 @@ public override bool Equals(object obj)
             return obj is Product other && Equals(other);
         }
 
-        public bool Equals(Product product)
+        public bool Equals(Product? product)
         {
-            return this.ProductType.Equals(product.ProductType)
-                   && this.NeutralMass.Equals(product.NeutralMass)
-                   && this.FragmentNumber == product.FragmentNumber
-                   && this.NeutralLoss.Equals(product.NeutralLoss)
-                   && this.SecondaryFragmentNumber == product.SecondaryFragmentNumber
-                   && this.SecondaryProductType == product.SecondaryProductType;
+            return product != null
+                   && NeutralMass.Equals(product.NeutralMass)
+                   && ProductType == product.ProductType
+                   && NeutralLoss.Equals(product.NeutralLoss)
+                   && Terminus == product.Terminus
+                   && FragmentNumber == product.FragmentNumber
+                   && ResiduePosition == product.ResiduePosition
+                   && SecondaryProductType == product.SecondaryProductType
```

Comment:
can we have null or empty variables in an equatable? what if no secondary product type.

Reply:
If no, it will check to see if the values are equivalent. The values could either be null or the default integer value. In the case where there is no secondary product time, it will check to see if null == null for that component

Reply:
is null == null?

Reply:
Yes, all null values are equal